src/features/pass-survey/model/hooks/useActivityState.ts

```diff
@@ -83,13 +83,41 @@ function useActivityState({
     }
   }
 
+  function setTimer(step: number, progress: number) {
+    if (!activityStorageRecord) {
+      return;
+    }
+
+    upsertActivityStorageRecord({
+      ...activityStorageRecord,
+
+      timers: {
```

Comment:
Although it is not important at the moment, I would recommend changing it right now this way:
...activityStorageRecord.timers,
[step]: progress,